Assets/Scripts/Player/PlayerControls.cs

```diff
@@ -60,6 +60,60 @@ public Jumping( PlayerActor player )
 		public override void Enter()
 		{
 			player.animator.SetBool( "isJumping", true );
+			player.physics.DoJump();
+		}
+
+		public override void Update()
+		{
+			if ( player.physics.GroundedCheck() )
+			{
+				if ( player.controls.jumpButton.down &&
+				     player.physics.JumpCheck() )
+				{
```

Comment:
This seems like a good place to actually execute our Jump if JumpCheck is true, right? Still seems a weird naming convention to me